### ISSUE TITLE
enhancement: a11y rules "click-events-have-key-events" and "no-static-element-interactions" fail with a warning. 

### ISSUE THREAD

**Issue by FatumaA** [open-sauced/app#4130]
### Describe the bug

[jsx-a11y eslint plugin](https://www.npmjs.com/package/eslint-plugin-jsx-a11y) was recently added to the project and these rules -jsx-a11y/click-events-have-key-events and jsx-a11y/no-static-element-interactions - fail with a warning.

This issue offers more context: https://github.com/open-sauced/app/issues/2264

### Steps to reproduce

The following components and files are causing the rule to fail:

1. components/atoms/PillSelector/pill-selector.tsx
2. components/atoms/Radio/radio.tsx
3. components/atoms/RadioCheck/radio-check.tsx
4. components/atoms/SVGIcon/svg-icon.tsx
5. components/atoms/Search/search.tsx
6. components/atoms/TextInput/text-input.tsx
7. components/atoms/ToggleOption/toggle-option.tsx
8. components/molecules/AuthSection/auth-section.tsx
9. components/molecules/CardRepoList/card-repo-list.tsx
10. components/molecules/ContributorFilterDropdown/contributor-filter-dropdown.tsx
11. components/molecules/ContributorHighlight/contributor-highlight-card.tsx
12. components/molecules/ContributorListTableRow/contributor-list-table-row.tsx
13. components/molecules/DevCard/dev-card.tsx
14. components/molecules/HighlightInput/highlight-input-form.tsx
15. components/molecules/HighlightsFeedCard/highlights-filter-card.tsx
16. components/molecules/InfoCard/info-card.tsx
17. components/molecules/RepoRow/repo-row.tsx
18. components/molecules/ReposoitoryCartItem/repository-cart-item.tsx
19. components/molecules/SuperlativeSelector/superlative-selector.tsx
20. components/organisms/DevCardWall/dev-card-wall.tsx
21. components/organisms/SearchDialog/search-dialog.tsx
22. pages/u/[username]/card.tsx

**Comment by FatumaA** [comment]
.take
